workflow/controller/workflowpod.go

```diff
@@ -747,3 +757,97 @@ func verifyResolvedVariables(obj interface{}) error {
 	})
 	return unresolvedErr
 }
+
+// Retrieve and create Volumes and Volumemount object for Pod
+func (woc *wfOperationCtx) createSecretVolumes(tmpl *wfv1.Template) (secretVolumes []apiv1.Volume, secretVolMounts []apiv1.VolumeMount) {
+	var allVolumesMap = make(map[string]apiv1.Volume)
+	var uniqueKeyMap = make(map[string]bool)
+
+	woc.createArgoArtifactsRepoSecret(tmpl, allVolumesMap, uniqueKeyMap)
+
+	for _, art := range tmpl.Outputs.Artifacts {
+		woc.createSecretVolume(allVolumesMap, art, uniqueKeyMap)
+	}
+	for _, art := range tmpl.Inputs.Artifacts {
+		woc.createSecretVolume(allVolumesMap, art, uniqueKeyMap)
+	}
+
+	if len(allVolumesMap) > 0 {
```

Comment:
The check for `len(allVolumesMap) > 0` is unnecessary. Golang can append to nil.